internal/plugin/execer.go

```diff
@@ -0,0 +1,136 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/hashicorp/go-argmapper"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/internal/funcspec"
+	"github.com/hashicorp/waypoint-plugin-sdk/internal/pluginargs"
+	pb "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+)
+
+// execerClient implements component.Execer for a service that
+// has the exec methods implemented.
+type execerClient struct {
+	Client  execerProtoClient
+	Logger  hclog.Logger
+	Broker  *plugin.GRPCBroker
+	Mappers []*argmapper.Func
+}
+
+func (c *execerClient) Implements(ctx context.Context) (bool, error) {
+	if c == nil {
+		return false, nil
+	}
+
+	resp, err := c.Client.IsExecer(ctx, &empty.Empty{})
+	if err != nil {
```

Comment:
I thinkkkkkk we need to detect a `codes.Unavailable` here (it might be a diff code) in the case we have a version mismatch between plugins and just return that as `false, nil`. 

Easiest way to figure this out would be to call an RPC against an endpoint that doesn't exist and see what happens and just use that code.